overlord/ifacestate/ifacemgr.go

```diff
@@ -77,6 +77,54 @@ func Manager(s *state.State, hookManager *hookstate.HookManager, extraInterfaces
 	return m, nil
 }
 
+func setInitialConnectAttributes(ts *state.Task, plugSnap string, plugName string, slotSnap string, slotName string) error {
+	// Set initial interface attributes for the plug- and slot- snaps in connect task.
+	var snapst snapstate.SnapState
+	var err error
+
+	st := ts.State()
+	if err = snapstate.Get(st, plugSnap, &snapst); err != nil {
+		return err
+	}
+
+	snapInfo, err := snapst.CurrentInfo()
+	if err != nil {
+		return err
+	}
+
+	plugAttrs := make(map[string]interface{})
+	if plug, ok := snapInfo.Plugs[plugName]; ok {
+		for k, v := range plug.Attrs {
+			plugAttrs[k] = v
+		}
+	} else {
+		return fmt.Errorf("Snap %q has no plug named %q", plugSnap, plugName)
+	}
+
+	if err = snapstate.Get(st, slotSnap, &snapst); err != nil {
+		return err
+	}
+	snapInfo, err = snapst.CurrentInfo()
+	if err != nil {
+		return err
+	}
+	snap.AddImplicitSlots(snapInfo)
```

Comment:
Curious. Why do we need this here?

Should we move this to a more central place?